tests/util_test.c

```diff
@@ -517,6 +518,30 @@ START_TEST(test_cc_oci_enable_networking) {
 
 } END_TEST
 
+START_TEST(test_dup_over_stdio) {
+	int fd = -1;
+	/* fd pointer is NULL */
+	ck_assert(! dup_over_stdio(NULL));
+
+	/* fd number is -1 */
+	fd = -1;
+	ck_assert(! dup_over_stdio(&fd));
+
+
+	/* dup not used but open STDIN fd */
+	fd = STDIN_FILENO ;
```

Comment:
This test is going to close whatever `STDIN_FILENO` is attached to. It should probably do something like the following to minimise surprises in the future if the test gets expanded [*]:

```
/* save */
int saved_stdin = dup(STDIN_FILENO).

<< run tests >>

/* restore */
dup2(STDIN_FILENO, saved_stdin);
```

(although of course if an assertion fires that won't help ;)

---

[*] - It should be ok to close stdin in a single `START_TEST/END_TEST` block as they are run in sub-processes.